### PR TITLE
add org.freedesktop.Sdk.Extension.deno

### PR DESCRIPTION
**Pull request by sigmaSd**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [ ] I have [built][build] and tested the submission locally. // WIP
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** // will do after I make sure it can work
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge). // I guess covered by the above
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: org.freedesktop.Sdk.Extension.deno.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+<component type="addon">
+  <id>org.freedesktop.Sdk.Extension.deno</id>
+  <extends>org.freedesktop.Sdk</extends>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>MIT</project_license>
+  <name>Deno</name>
+  <summary>JavaScript runtime built on V8 JavaScript engine</summary>
+  <url type="homepage">https://deno.com/</url>
+  <releases>
+  </releases>
```
Comment (bbhtt) [inline org.freedesktop.Sdk.Extension.deno.metainfo.xml:11]: Why is this empty?

File: org.freedesktop.Sdk.Extension.deno.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="addon">
+  <id>org.freedesktop.Sdk.Extension.deno</id>
+  <extends>org.freedesktop.Sdk</extends>
+  <metadata_license>CC0-1.0</metadata_license>
```
Comment (bbhtt) [inline org.freedesktop.Sdk.Extension.deno.metainfo.xml:5]: Add developer_name

File: org.freedesktop.Sdk.Extension.deno.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        only-arches:
+          - aarch64
+      - type: archive
+        url: https://github.com/denoland/deno/releases/download/v1.38.4/deno_src.tar.gz
+        sha256: a041d7407d89b94c7d2d863397159799c44ea5012eb3ff6870277003620b3ef4
+      - cargo-sources.json
+    build-options:
+      append-path: /usr/lib/sdk/rust-stable/bin
+      env:
+        CARGO_HOME: /run/build/deno/cargo
```
Comment (bbhtt) [inline org.freedesktop.Sdk.Extension.deno.yaml:61]: Move this on before buildsystem.

**Comment by sigmaSd** [comment]
bot, build org.freedesktop.Sdk.Extension.deno

File: org.freedesktop.Sdk.Extension.deno.yaml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.rust-stable
+build-extension: true
+separate-locales: false
+appstream-compose: false
+build-options:
+ prefix: /usr/lib/sdk/deno
+ env:
+    PATH: /usr/lib/sdk/deno/bin:/app/bin:/usr/bin
+modules:
```
Comment (bbhtt) [inline org.freedesktop.Sdk.Extension.deno.yaml:15]: You did not install the appdata

File: org.freedesktop.Sdk.Extension.deno.yaml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  # Required for deno to build
+  - name: protobuf
+    buildsystem: cmake-ninja
+    builddir: true
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=None
+      - -DBUILD_SHARED_LIBS=ON
```
Comment (bbhtt) [inline org.freedesktop.Sdk.Extension.deno.yaml:22]: Why enable shared?

Comment (sigmaSd) [reply org.freedesktop.Sdk.Extension.deno.yaml:22]: no idea, I copied protobuf compilation from somewhere else, I'll try without it

File: org.freedesktop.Sdk.Extension.deno.metainfo.xml (hunk 2 above)
Comment (sigmaSd) [reply org.freedesktop.Sdk.Extension.deno.metainfo.xml:5]: done

File: org.freedesktop.Sdk.Extension.deno.yaml (hunk 4 above)
Comment (sigmaSd) [reply org.freedesktop.Sdk.Extension.deno.yaml:15]: I added `      - install -Dm644 ${FLATPAK_ID}.metainfo.xml -t ${FLATPAK_DEST}/share/metainfo` is that correct ?

**Comment by sigmaSd** [comment]
Maybe people want to put their deno apps on flathub (probably based on fresh and webview /electron) , but there is no deno so no one bother. Same for node ?
But I admit this is just hypothetical.

I was trying just to move forward https://github.com/flathub/flathub/pull/4742 do you think that pr have a chance of landing without something like this ?

**Pull request closed without merging**